oracle/oracles/pd.go

```diff
@@ -165,61 +169,49 @@ func (o *pdOracle) setLastTS(ts uint64, txnScope string) {
 	}
 	lastTSInterface, ok := o.lastTSMap.Load(txnScope)
 	if !ok {
-		lastTSInterface, _ = o.lastTSMap.LoadOrStore(txnScope, new(uint64))
+		lastTSInterface, _ = o.lastTSMap.LoadOrStore(txnScope, &atomic.Pointer[lastTSO]{})
 	}
-	lastTSPointer := lastTSInterface.(*uint64)
+	lastTSPointer := lastTSInterface.(*atomic.Pointer[lastTSO])
+	var current *lastTSO
 	for {
-		lastTS := atomic.LoadUint64(lastTSPointer)
-		if ts <= lastTS {
+		last := lastTSPointer.Load()
+		if last != nil && ts <= last.tso {
```

Comment:
Is it ok to put a nil value into the `lastTSPointer`? How about using a constant value instead of nil?

Reply:
Made it a non-nil pointer.